ocaml/networkd/bin/network_server.ml

```diff
@@ -35,19 +35,106 @@ let write_config () =
     try Network_config.write_config !config
     with Network_config.Write_error -> ()
 
+let get_index_from_ethx name =
+  try Scanf.sscanf name "eth%d%!" Option.some with _ -> None
+
+let sort_based_on_ethx () =
+  Sysfs.list ()
+  |> List.filter_map (fun name ->
+         if Sysfs.is_physical name then
+           get_index_from_ethx name |> Option.map (fun i -> (name, i))
+         else
+           None
+     )
+
+let read_previous_inventory () =
+  let previous_inventory = "/var/tmp/.previousInventory" in
+  try
+    Xapi_stdext_unix.Unixext.file_lines_fold
+      (fun acc line ->
+        match Inventory.parse_inventory_entry line with
+        | Some ("MANAGEMENT_INTERFACE", iface) ->
+            info "get management interface from previous inventory: %s" iface ;
+            (iface, snd acc)
+        | Some ("MANAGEMENT_ADDRESS_TYPE", addr_type) ->
+            info "get management address type from previous inventory: %s"
+              addr_type ;
+            (fst acc, addr_type)
+        | _ ->
+            acc
+      )
+      ("", "") previous_inventory
+  with e ->
+    error "Failed to read previous inventory %s: %s" previous_inventory
+      (Printexc.to_string e) ;
+    raise
+      (Network_error
```

Comment:
I'd like to see an easier way to raise these if it does not already exist. More like:
```
let internal_error fmt =
  Printf.ksprintf
    (fun str ->
      error "%s" str ;
      raise Api_errors.(Server_error (internal_error, [str]))
    )
    fmt

```